src/ray_serializer.jl

```diff
@@ -52,4 +52,20 @@ function serialize_to_bytes(x)
     return bytes
 end
 
+function serialize_to_ray_object(x)
+    bytes = Vector{UInt8}()
+    s = RaySerializer(bytes)
+    writeheader(s)
+    serialize(s, x)
+
+    buffer = ray_jll.LocalMemoryBuffer(bytes, sizeof(bytes), true)
+    metadata = ray_jll.NullPtr(ray_jll.Buffer)
```

Comment:
do we know yet what metadata is supposed to hold?

Reply:
we have some hunches yes :)  centralizing the "serialize this thing to a RayObject" in a single function means there's a centralized place where we can handle the metadata too, but we're not yet at a point where we can tackle that.